pkg/scheduler/schedule_one.go

```diff
@@ -843,17 +841,22 @@ func truncateMessage(message string) string {
 	return message[:max-len(suffix)] + suffix
 }
 
-func updatePod(client clientset.Interface, pod *v1.Pod, condition *v1.PodCondition, nominatingInfo *framework.NominatingInfo) error {
+func updatePod(client clientset.Interface, pod *v1.Pod, condition *v1.PodCondition, nominatingInfo *framework.NominatingInfo) {
 	klog.V(3).InfoS("Updating pod condition", "pod", klog.KObj(pod), "conditionType", condition.Type, "conditionStatus", condition.Status, "conditionReason", condition.Reason)
 	podStatusCopy := pod.Status.DeepCopy()
 	// NominatedNodeName is updated only if we are trying to set it, and the value is
 	// different from the existing one.
 	nnnNeedsUpdate := nominatingInfo.Mode() == framework.ModeOverride && pod.Status.NominatedNodeName != nominatingInfo.NominatedNodeName
 	if !podutil.UpdatePodCondition(podStatusCopy, condition) && !nnnNeedsUpdate {
-		return nil
+		return
 	}
 	if nnnNeedsUpdate {
 		podStatusCopy.NominatedNodeName = nominatingInfo.NominatedNodeName
 	}
-	return util.PatchPodStatus(client, pod, podStatusCopy)
+	go func() {
+		// update Pod's status in goroutine because there might be retries.
+		if err := util.PatchPodStatus(client, pod, podStatusCopy); err != nil {
+			klog.ErrorS(err, "Error updating pod", "pod", klog.KObj(pod))
```

Comment:
```suggestion
			klog.ErrorS(err, "Could not update pod status", "pod", klog.KObj(pod))
```

Reply:
or maybe we could even say: "after scheduling attempt"? Are there other calls to `updatePod` that are not related to the scheduling attempt?

Reply:
> Are there other calls to updatePod that are not related to the scheduling attempt?

Seems no. So adding "after scheduling attempt" also looks good. 👍